reactor-netty-http/src/main/java/reactor/netty/http/client/MicrometerHttp2ConnectionProviderMeterRegistrar.java

```diff
@@ -59,4 +60,13 @@ void registerMetrics(String poolName, String id, SocketAddress remoteAddress, In
 		     .tags(tags)
 		     .register(REGISTRY);
 	}
+
+	void deRegisterMetrics(String poolName, String id, SocketAddress remoteAddress) {
```

Comment:
In `Http2ConnectionProvider` we need to override `reactor.netty.resources.PooledConnectionProvider#deRegisterDefaultMetrics`

```
	@Override
	protected void deRegisterDefaultMetrics(String id, SocketAddress remoteAddress) {
		MicrometerHttp2ConnectionProviderMeterRegistrar.INSTANCE
				.deRegisterMetrics(name(), id, remoteAddress);
	}
```